test/e2e/create_enable_disable_delete.go

```diff
@@ -12,68 +12,186 @@ import (
-		verifier.verifyThatLogsAreNotSentToEchoServer(ctx, "other-program", "1", "this should not get sent to echo server")
-		verifier.verifyThatLogsAreNotSentToEchoServer(ctx, "test-program", "3", "this should not get sent to echo server")
-
-		By("Disable the shoot-rsyslog-relp extension")
-		ctx, cancel = context.WithTimeout(parentCtx, 15*time.Minute)
-		defer cancel()
-		Expect(f.UpdateShoot(ctx, f.Shoot, func(shoot *gardencorev1beta1.Shoot) error {
-			for i, extension := range shoot.Spec.Extensions {
-				if extension.Type == "shoot-rsyslog-relp" {
-					shoot.Spec.Extensions = append(shoot.Spec.Extensions[:i], shoot.Spec.Extensions[i+1:]...)
+			By("Verify shoot-rsyslog-relp works")
+			ctx, cancel = context.WithTimeout(parentCtx, 5*time.Minute)
+			DeferCleanup(cancel)
+
+			verifier, err := newVerifier(ctx, f.Logger, f.ShootFramework.SeedClient, f.ShootFramework.ShootSeedNamespace(), "local", f.Shoot.Name, string(f.Shoot.UID))
+			Expect(err).NotTo(HaveOccurred())
+
+			verifier.verifyThatLogsAreSentToEchoServer(ctx, "test-program", "1", "this should get sent to echo server")
+			verifier.verifyThatLogsAreNotSentToEchoServer(ctx, "other-program", "1", "this should not get sent to echo server")
+			verifier.verifyThatLogsAreNotSentToEchoServer(ctx, "test-program", "3", "this should not get sent to echo server")
+
+			By("Disable the shoot-rsyslog-relp extension")
+			ctx, cancel = context.WithTimeout(parentCtx, 15*time.Minute)
+			DeferCleanup(cancel)
+			Expect(f.UpdateShoot(ctx, f.Shoot, func(shoot *gardencorev1beta1.Shoot) error {
+				for i, extension := range shoot.Spec.Extensions {
+					if extension.Type == "shoot-rsyslog-relp" {
+						shoot.Spec.Extensions = append(shoot.Spec.Extensions[:i], shoot.Spec.Extensions[i+1:]...)
+					}
 				}
+				return nil
+			})).To(Succeed())
+
+			By("Verify that shoot-rsyslog-relp extension is disabled")
+			ctx, cancel = context.WithTimeout(parentCtx, 5*time.Minute)
+			DeferCleanup(cancel)
+			verifier.verifyThatLogsAreNotSentToEchoServer(ctx, "test-program", "1", "this should not get sent to echo server")
+
+			By("Delete Shoot")
+			ctx, cancel = context.WithTimeout(parentCtx, 15*time.Minute)
+			DeferCleanup(cancel)
+			Expect(f.DeleteShootAndWaitForDeletion(ctx, f.Shoot)).To(Succeed())
+
+			By("Delete additional garden cluster resources")
+			for _, obj := range gardenClusterResources {
+				Expect(f.GardenClient.Client().Delete(ctx, obj)).To(Succeed())
+				f.Logger.Info("Deleted resource", "resource", client.ObjectKeyFromObject(obj))
 			}
-			return nil
-		})).To(Succeed())
+		})
+	}
 
-		By("Verify that shoot-rsyslog-relp extension is disabled")
-		ctx, cancel = context.WithTimeout(parentCtx, 5*time.Minute)
-		defer cancel()
-		verifier.verifyThatLogsAreNotSentToEchoServer(ctx, "test-program", "1", "this should not get sent to echo server")
+	Context("shoot-rsyslog-relp extension with tls disabled", Label("tls-disabled"), func() {
```

Comment:
Idea: instead of passing a Shoot and mutate func and gardenClusterResources to be created, you could make the `test` func a helper func (without the IT inside). The test func can accept the final Shoot and you can set up the needed Shoot and created the needed resources a BeforeEach or as first step in ITs. WDYT?

Reply:
I wanted to test the enablement and disablement of the extension in an already created shoot, that's why I use the mutate func.

If the `test()` func does not contain an `It()` but the other way around, I would have to code it like this:

```go
Context("shoot-rsyslog-relp extension with tls disabled", func() {
  	f := defaultShootCreationFramework()
	f.Shoot = e2e.DefaultShoot("e2e-rsyslog-relp")
	
	It("Create Shoot, enable shoot-rsyslog-relp extension then disable it and delete Shoot", func() {
		test(f, ...)
        })   
})

Context("shoot-rsyslog-relp extension with tls disabled", func() {
  	f := defaultShootCreationFramework()
	f.Shoot = e2e.DefaultShoot("e2e-rsyslog-relp")
	BeforeEach(func() {
                //create resources
        })

	It("Create Shoot, enable shoot-rsyslog-relp extension then disable it and delete Shoot", func() {
                 // optionally create resources here
		test(f, ...)
        })   
})
```

If we want to run the 2 cases in Parallel, we need to use 2 separate `ShootCreationFrameWork`s. Additionally, the  `defaultShootCreationFramework()` function will call a few `BeforeEach()` functions so it cannot be used within an `It()`.

Initially I had it coded like this:
```go
test = func(f *ShootCreationFramework) {
	It("Create Shoot, enable shoot-rsyslog-relp extension then disable it and delete Shoot", func() {...})
}

Context("shoot-rsyslog-relp extension with tls disabled", func() {
  	f := defaultShootCreationFramework()
	f.Shoot = e2e.DefaultShoot("e2e-rsyslog-relp")
	
	test(f, mutateFunc)
})

Context("shoot-rsyslog-relp extension with tls disabled", func() {
  	f := defaultShootCreationFramework()
	f.Shoot = e2e.DefaultShoot("e2e-rsyslog-relp")
	BeforeEach(func() {
                // We need access to `f` here to create resources
        })

	test(f, mutateFunc)  
})
```

But I wanted to avoid the duplicated code for the creation of the `ShootCreationFramework`